### PR TITLE
Add support for Python 3

### PR DESCRIPTION
**Pull request by diox**
Supersedes #8

fa5

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,7 @@
 .*.sw?
 *.pyc
 *.pyo
+.cache/
 build/
 dist/
+spidermonkey/
```
Comment (EnTeQuAk) [inline .gitignore]: not sure why this is here but that folder should not be ignored imho.

File: .travis.yml [hunk 2]
```diff
@@ -2,6 +2,7 @@ language: python
 sudo: false
 python:
 - '2.7'
+- '3.5'
```
Comment (EnTeQuAk) [inline .travis.yml]: Can we add 3.6 here too?

File: setup.py [hunk 3]
```diff
@@ -35,6 +35,10 @@ def is_pure(self):
         'Development Status :: 4 - Beta',
         'Intended Audience :: Developers',
         'License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)',
-        'Topic :: Software Development :: Libraries :: Python Modules',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.5',
```
Comment (EnTeQuAk) [inline setup.py]: Same here, let's test python 3.6 too

File: .gitignore (hunk 1 above)
Comment (diox) [inline .gitignore]: By default `make download` puts the spidermonkey it downloads there.

**Comment by EnTeQuAk** [comment]
@diox do you have time to follow up on this? Otherwise I can pick it up. I think we eventually agreed to just port the damn validator and be done with it :)

**Comment by diox** [comment]
I don't have time to pick it up right now, so if you do, please go ahead :)

**Comment by EnTeQuAk** [comment]
Closing in favor of https://github.com/mozilla/python-spidermonkey/pull/12

**Pull request closed without merging**